.ruff.toml

```diff
@@ -0,0 +1,14 @@
+extend-exclude = [
+  "**/renv",
+  "**/.venv",
+  ".github",
+  ".vscode",
+  "**/node_modules",
+  "**/__pycache__",
+  "**/build",
+  # file with cell magic not parsed by ruff
+  "src/resources/jupyter/lang/python/cleanup.py",
+  "src/resources/jupyter/lang/python/setup.py",
```

Comment:
@cscheid we already don't process those files as they don't play well with **ruff** as you guessed. This why they are not formatted in this PR.